### ISSUE TITLE
Should undocumented figure fill/outline in subset be removed?

### ISSUE THREAD

**Issue by PaulWessel** [GenericMappingTools/gmt#1342]
**Description of the problem**

During implementation I added various debug-related modifiers to **-F** in subplot:

- **+d** draws the rectangles around what is considered the total dimensions of each subplot (this includes area computed to hold ticks, annotations)
- **+g** sets fill for the total composite figure domain
- **+p** sets pen to draw outline of total composite figure domain.

For instance, this code:

```
gmt begin panels
  gmt subplot begin 2x2 -Fs3i+glightgray+pfaint+d -M5p -A -SCt -SR -Bwsne
    gmt basemap -R0/80/0/10
    gmt basemap -c
    gmt basemap -c
    gmt basemap -c
  gmt subplot end
gmt end show
```

yields the plot below  My question: is either **+g** or **+p** useful to users and should be documented? I cannot see the point of **+p**, but perhaps **+g**?

![panels](https://user-images.githubusercontent.com/26473567/62748682-36430a00-b9f5-11e9-9fd8-695308b8ab19.png)

**Comment by seisman** [comment]
I don't think any of these three modifiers are useful to users. I suggest keeping it as is.

**Comment by PaulWessel** [comment]
I agree, will keep them hidden for developers only.

**Comment by joa-quim** [comment]
Guys, you were too quick on this one. I think the **+d** could be interesting in some cases. Imagine for example a portfolio of global map projections, each one seating in its own box. But it should be expanded to **+d<pen>** (didn't get what **+p** does as it seem to do what I understood **+d** would do, except the red).

**Comment by seisman** [comment]
**+d** draws the box for each subplot, and **+p** draws the box of the whole figure.

**Comment by PaulWessel** [comment]
The +d is debug and will not be revealed to users.  For +p and +g, I can see uses to paint and outline the background when no annotations and frames are used internally.  Consider this plot:
![four](https://user-images.githubusercontent.com/26473567/62803600-9930b180-ba86-11e9-9eff-179b98823468.png)

[Strange, for me it truncates the bottom but the plot is correct].  The script is

```
gmt begin four png
  gmt subplot begin 2x2 -Fs8c+p1p+gyellow -B+n -Vd -M0
    gmt pscoast -Rg -JG0/0/? -Gred -Sblue -B0g
    gmt pscoast -JG90/0/? -Gred -Sblue -B0g -c
    gmt pscoast -JG180/0/? -Gred -Sblue -B0g -c
    gmt pscoast -JG270/0/? -Gred -Sblue -B0g -c
  gmt subplot end
gmt end show
```

**Comment by seisman** [comment]
Looks good. For such cases, we probably need a +cdx/dy modifier to specify extra clearances between the figure margin and the surrounding box.

**Comment by PaulWessel** [comment]
I think that is OK as long as that is a separate thing from the **-R** I create, i.e., the box will just bleed further out but the logical lower left corner is fixed.  That is what you mean, yes?

**Comment by seisman** [comment]
Yes.

**Comment by joa-quim** [comment]
Sorry, have the house full so can't keep the pace with this discussion.
When I add +d it seems to reveal a cm vs inc problem. What I was thinking is an option that plots a rectangle around each subplot. We can construct one with -B for cartesians but not for  other projections.

**Comment by PaulWessel** [comment]
OK, post that when you get a chance.  Make sure you update as I fixed something like that yesterday.

**Comment by joa-quim** [comment]
This is not particularly important and may wait till 6.1 but I think it should allow to plot frames only around filled subplots, but for it the +p +g must be a per panel feature.
```
gmt begin four png
  gmt subplot begin 3x3 -Fs8c+d -B+n -M0.5
    gmt pscoast -Rg -JG0/0/? -Gred -Sblue -B0g
    gmt pscoast -JG90/0/? -Gred -Sblue -B0g -c1,1
    gmt pscoast -JG180/0/? -Gred -Sblue -B0g -c2,2
  gmt subplot end
gmt end show
```

![four](https://user-images.githubusercontent.com/537321/62824436-793be380-bb95-11e9-9514-59cf1533a241.png)

**Comment by PaulWessel** [comment]
Maybe a **-F** modifier **+i** to let **+g**, **+p** apply to individual canvases (all 9 in your car) and not the entire figure, and **+I** to only those that were in fact selected (3 in your case)?  I agree this would be an enhancement that can wait until 6.1.

**Comment by joa-quim** [comment]
Yes, close to that but let **+I** be a **-I** of *gmt subplot set* so that one can change bg color etc per subplot.
